cachi2/interface/cli.py

```diff
@@ -97,14 +96,57 @@ def _looks_like_json(value: str) -> bool:
     return value.lstrip().startswith(("{", "["))
 
 
+class _Input:
+    packages: list[dict]
+    flags: list[str] = list()
+
+    def __init__(self, input: str) -> None:
+        if _looks_like_json(input):
+            self._parse_json(input)
+        else:
+            self._parse_packages(input)
+
+    def _parse_json(self, input: str) -> None:
+        input_obj = json.loads(input)
+
+        if isinstance(input_obj, dict):
+            if "packages" in input_obj.keys():
+                self._parse_packages(input_obj["packages"])
+            else:
+                self._parse_packages(input_obj)
+
+            if "flags" in input_obj.keys():
+                self._parse_flags(input_obj["flags"])
+        else:
+            self._parse_packages(input_obj)
```

Comment:
Do we want the format to be _that_ loose? I think the examples in the commit message could reasonably be the only allowed formats

This doesn't feel like it should be valid
```
{"packages": "pip"}
```

Reply:
My reasoning was that "packages" would be whatever could be read by the previous --package param. But fair point, we can stick to "if you define a packages key, it needs to be a dict list".

Reply:
Yeah, I would prefer an easier-to-define format

With the current one, you can get some really cryptic errors

```
$ cachi2 fetch-deps '{"flags": "foo"}'    
2023-01-04 14:42:09,863 ERROR InvalidInput: 2 validation errors for user input\npackages -> 0\n  Discriminator 'type' is missing in value (discriminator_key=type)\nflags -> 0\n  unexpected value; permitted: 'cgo-disable', 'force-gomod-tidy', 'gomod-vendor', 'gomod-vendor-check' (given=foo; permitted=('cgo-disable', 'force-gomod-tidy', 'gomod-vendor', 'gomod-vendor-check'))
Error: InvalidInput: 2 validation errors for user input
packages -> 0
  Discriminator 'type' is missing in value (discriminator_key=type)
flags -> 0
  unexpected value; permitted: 'cgo-disable', 'force-gomod-tidy', 'gomod-vendor', 'gomod-vendor-check' (given=foo; permitted=('cgo-disable', 'force-gomod-tidy', 'gomod-vendor', 'gomod-vendor-check'))
```

Reply:
Changed it.